model_analyzer/config/generate/run_config_generator.py

```diff
@@ -55,17 +58,25 @@ def next_config(self):
         RunConfig
             The next RunConfig generated by this class
         """
-        if self._curr_pacg.is_done():
-            self._curr_model_config = self._curr_mcg.next_config()
-            self._curr_pacg = PerfAnalyzerConfigGenerator(
-                self._config, self._curr_model_config.get_field("name"),
-                self._model_pa_flags, self._model_parameters)
 
-        perf_analyzer_config = self._curr_pacg.next_config()
-        run_config = self._generate_run_config(self._curr_model_config,
-                                               perf_analyzer_config)
+        pacg = PerfAnalyzerConfigGenerator(self._config, self._model_name,
+                                           self._model_pa_flags,
+                                           self._model_parameters)
 
-        return run_config
+        while not pacg.is_done():
+            perf_analyzer_config = pacg.next_config()
+
+            model_configs = self._generate_all_model_config_permuations(
+                self._models)
+
+            for model_config in model_configs:
```

Comment:
this isn't really a model_config, right? Is it a list of model_configs?

Reply:
Correct

Reply:
I've changed the name to reflect this